change-notes/1.20/analysis-cpp.md

```diff
@@ -20,6 +20,7 @@
 | Lossy function result cast (`cpp/lossy-function-result-cast`) | Fewer false positive results | The whitelist of rounding functions built into this query has been expanded. |
 | Unused static variable (`cpp/unused-static-variable`) | Fewer false positive results | Variables with the attribute `unused` are now excluded from the query. |
 | Resource not released in destructor (`cpp/resource-not-released-in-destructor`) | Fewer false positive results | Fix false positives where a resource is released via a virtual method call, function pointer, or lambda. |
+| User-controlled data in arithmetic expression (`cpp/tainted-arithmetic`) | More correct results | This query has been reworked so that it can find a wider variety of results. |
```

Comment:
I've forgotten the procedure for writing change notes for queries that aren't on LGTM. Don't we keep them separate from the LGTM queries?

Reply:
We haven't really discussed how to treat non-LGTM change notes recently. There are still change notes in the other repository, it's just that no-one's created any for 1.20 yet. For 1.19, people used the other notes for queries like this and for extractor changes (since it meant that they could continue to include the change notes in the relevant PR). The resulting two files were then combined to give:

* LGTM only information: https://lgtm.internal.semmle.com/help/lgtm/changes/lgtm-cpp
* Full information: https://help.semmle.com/wiki/pages/viewpage.action?pageId=29394282

Looking at this 1.20 change notes file, I can see that people aren't always including the expected information for new queries. It sounds as if we need to agree new guidelines, that work for the current workflow and also provide the information we need for users.

Reply:
@geoffw0 It sounds like this change note belongs in the internal repo, then.

Reply:
OK, I'll move it to the internal repo - but I note that it's a pain having change notes from QL in a different repo.

Reply:
Done (https://git.semmle.com/Semmle/code/pull/30344).